### ISSUE TITLE
Ubuntu: Keybase stuck when inputting anything into the GUI login

### ISSUE THREAD

**Issue by ehteqx** [keybase/client#4097]
When starting the Keybase GUI in Ubuntu - whether it is because of autostart, manual launch from the H.U.D. or directly from the terminal, the main windows never shows up (the application starts minimized) and the user is not logged in.

If opening the GUI (left click --> open keybase) and trying to log in, the GUI windows shows up, but it starts an endless (seemingly) loading process after anything is inserted into the text-fields (password, other username, even blanks) and trying to proceed.

DETAILS:
Operative System: Ubuntu 14.04 LTS, x86_64 with Unity;
Keybase GUI Version: 1.0.17-20160825160033+cefe001

P.S.: I am not able to verify if the problem is reproduceable, because I temporarily lack another similar installation to try. I hope the info I provided is enough. Thanks!

**Comment by cjb** [comment]
Hi, please could you run `keybase log send` and tell us the ID? Thanks!

**Comment by ehteqx** [comment]
Hello! I kept experimenting, and I foud out that triggering a login from the terminal while the GUI is still running, returns the message "Error: server EOF" both on the GUI windows and the terminal, whereas a simple shell login while the GUI is not running succeeds in the process. Opening the GUI at this stage triggers a password input prompt due to KBFS unlock and everything works fine.

I have sent the log, the ID is: a91466880597464dca4bc11c
Please, note that now (at the moment of log submission) the application is working due to the workaroud above.

**Comment by cjb** [comment]
Thanks!

@maxtaco @patrickxb, there is a service SIGSEGV panic in this log send.

**Comment by patrickxb** [comment]
Sorry for the issue, @ehteqx ...

It looks like the service is crashing on a nil pointer, but I'm not sure why:

```
	if s != nil {
		storedSecretUsernames, err = s.GetUsersWithStoredSecrets()
	}
```

```
func (s *SecretStoreLocked) GetUsersWithStoredSecrets() ([]string, error) {
	s.Lock()   // crashing here
	defer s.Unlock()
	return s.SecretStoreAll.GetUsersWithStoredSecrets()
}
```

The `!= nil` check was intended to protect against this crash, and all the tests pass.

Still looking into it...thanks for your patience.

**Comment by maxtaco** [comment]
Is it the `nil` `interface` versus typed pointer bug?

**Comment by patrickxb** [comment]
probably....

**Comment by gabriel** [comment]
Yeah we had a logger issue that was similar awhile back (reciever can be nil)

**Comment by gabriel** [comment]
```go
func (s *SecretStoreLocked) GetUsersWithStoredSecrets() ([]string, error) {
  if s == nil {
    return nil, fmt.Errorf("Nil secret store")
  }
}
```

**Comment by cjb** [comment]
I wonder why it's only affecting one user, though.

**Comment by ehteqx** [comment]
Hello! @patrickxb, absolutely no problem. Your service is great and the issue is not causing anything more than a slight discomfort at startup... Anyway I just wanted to let you know.

I don't know if it can help, but trying to quit Keybase from the GUI and then re-launching it seems not to reproduce the problem (the KBFS unlock password is required, and then all works fine), but a reboot does.

**Comment by ehteqx** [comment]
Hello everybody! With the new release, the problem seems to be fixed to me. Thanks for the support and the rapid fix. If it's OK you you, I think I can close the thread.
Have a nice day!

**Comment by keeGAsIydnA** [comment]
Hi I seem to be having the same problem as described here. I'm running the latest build when following your instructions here... https://keybase.io/docs/the_app/install_linux
It just hangs and says loading. I've tried launching both from gui and terminal. I also tried moving /usr/bin/keybase to another location as described in another thread. Didn't work so I moved back. Still no luck. Any guidance? It works fine for me on the Windows App.
